django_declarative_apis/machinery/tasks.py

```diff
@@ -73,19 +69,18 @@ def _log_task_stats(
 
         cache.set(QUEUE_LENGTH_CACHE_KEY, queue_length)
 
-        if newrelic_agent:
-            newrelic_agent.record_custom_event(
-                "task_runner:queue_length",
-                {
-                    "queue_length": queue_length,
-                    "queue": queue,
-                    "routing_key": routing_key,
-                    "wait_time_seconds": wait_time,
-                    "queue_delay_seconds": queue_delay,
-                    "process_task_count": process_task_count,
-                    "correlation_id": correlation_id,
-                },
-            )
+        emit_events(
+            EventType.QUEUE_LENGTH.value,
```

Comment:
this is a little odd. the current value of the enum is the value expected by newrelic schema. if we're removing that from the library, i'd expect `emit_events` to take an enum member rather than the concrete value. that way, callers can use the enum members directly for internal mapping rather than value. not merge blocking though.

Reply:
also, the event name should probably be something like `QUEUE_SNAPSHOT` or something along those lines. length is one of the attributes and doesn't really represent the payload.